Actually execute all proofs as part of CI

dconnolly opened the pull request:
Hi, I noticed that not all of the proof examples are being parsed and checked as part of regular CI, I created a workflow to do so - several are broken:

<img width="825" alt="image" src="https://github.com/ProofFrog/ProofFrog/assets/552961/b058a306-35c3-40f1-8831-1c6efb45ca2a">

Qcode commented:
Hi Deidre,

Thanks for this contribution! Some of the proofs in the examples directory were aspirational, they were initially developed as we were speccing out of our AST but ended up being out of scope for what could be accomplished during the masters. I have added a .todo extension so they are not picked up by your globbing pattern. Furthermore, I noticed that the CI actually passed for this test run. So I also added an exit code to the testProofs script which didn't exist before. I believe this should give us failing CI for any broken proofs or unit tests, and only test proofs that we have managed to get working in the engine so far.

dconnolly commented:
> Hi Deirdre,
> 
> Thanks for this contribution! Some of the proofs in the examples directory were aspirational, they were initially developed as we were speccing out of our AST but ended up being out of scope for what could be accomplished during the masters. I have added a .todo extension so they are not picked up by your globbing pattern. Furthermore, I noticed that the CI actually passed for this test run. So I also added an exit code to the testProofs script which didn't exist before. I believe this should give us failing CI for any broken proofs or unit tests, and only test proofs that we have managed to get working in the engine so far.

Fantastic, thank you!
